### PR TITLE
[incubator/mysqlha] According to replicas count to create services for every pod

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: Aisuko <urakiny@gmail.com>

#### What this PR does / why we need it:
          
 This feature achieved that when the `Service Kind ` is NodePort. User can via every pod's service to connect to the pod and would single get metric container's log on every pod. And the services are the same ordering with the pods.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

**Comment by Aisuko** [comment]
Could I get any  👀 on this? It has been more than a month with no feedback here.   @prydonius 

cc: @maorfr You approve the last PR into this chart, any feedback that you can give would be helpful.

**Comment by maorfr** [comment]
/unassign @prydonius 
/assign @jpoon 

this may not work until @jpoon accepts his invite to be a trusted collaborator.
but feel free to collaborate on this chart from now on! ;)

File: incubator/mysqlha/templates/NOTES.txt [hunk 1]
```diff
@@ -17,3 +17,17 @@ To connect to your database:
 2. Open a connection to one of the MySQL pods
 
     mysql -h {{ template "fullname" . }}-0.{{ template "fullname" . }} -p
+
+{{- if .Values.service.type }}
```
Comment (jpoon) [inline incubator/mysqlha/templates/NOTES.txt:21]: does this need to be service.type = nodeport? or add specific documentation for other service types?

Comment (Aisuko) [reply incubator/mysqlha/templates/NOTES.txt:21]: I added specific `service.Type` logic to this. According to `service.Type` can print different information to the screen

File: incubator/mysqlha/values.yaml [hunk 2]
```diff
@@ -82,3 +82,11 @@ metrics:
     requests:
       cpu: 100m
       memory: 100Mi
+
+
+service:
+  ## Specify the nodePort value for the LoadBalancer and NodePort service types.
```
Comment (jpoon) [inline incubator/mysqlha/values.yaml:88]: nit: Is this comment necessary as it's just the definition of a node port?

File: incubator/mysqlha/templates/svc.yaml [hunk 3]
```diff
@@ -1,44 +1,41 @@
...
+{{- $fullname := include "fullname" . }}
+{{- $rc_number := int .Values.mysqlha.replicaCount }}
+{{- $root := . }}
+{{ range $i := until $rc_number }}
+---
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ template "fullname" . }}-readonly
+  name: {{ template "fullname" $root }}-svc-{{ $i }}
```
Comment (jpoon) [inline incubator/mysqlha/templates/svc.yaml:11]: The intent is to connect to this service only for reads as writes always need to go to the primary (mysql-0). As such, I think we should keep `readonly` in the name as well as the above comment explaining this.

File: incubator/mysqlha/values.yaml (hunk 2 above)
Comment (Aisuko) [reply incubator/mysqlha/values.yaml:88]: Oh, It's my fault, I will be careful more.

File: incubator/mysqlha/templates/svc.yaml (hunk 3 above)
Comment (Aisuko) [reply incubator/mysqlha/templates/svc.yaml:11]: Your mean like below? 

```name: {{ template "fullname" $root }}-readonly-{{ $i }}```

**Comment by Aisuko** [comment]
@jpoon Thanks a lot for your time to help me review, but I think maybe I had a wrong way to understand the service, and I want to make more sense of the service. 😃

**Pull request closed without merging**